### DISCUSSION TITLE
What type of sound set is used in represent instrument programs in tracks?

### DISCUSSION THREAD

**Discussion by TheLucifurry** [CoderLine/alphaTab#924] [Q&A] [answered]
Now, I learn specifications of sound sets  (e.g. https://en.wikipedia.org/wiki/Comparison_of_MIDI_standards)
And it seems, that alphatab uses the [`General MIDI 1`](https://www.midi.org/specifications-old/item/gm-level-1-sound-set) set, but maybe it uses another one, and i want to know exactly

**Comment by Danielku15** [comment, marked as answer]
On high level I would say yes. alphaTab uses the normal General Midi instrument set. 

In detail: alphaTab itself does not handle much specifics in regards of midi beside holding a number in the data model. When reading files like Guitar Pro or MusicXML the information in this file is just placed in the data model and used when generating the midi events. 

In theory: If you put the midi program 0x01 (Bright Piano) into your MusicXML alphaTab will also hold 0x01 in the `track.PlaybackInformation.Program` and write it into the MIDI Program Change events. The synthesizer loads then the SoundFont file and will select also from this file the program 1. If in this SoundFont2 file the author decided to put an Acoustic Guitar to program 01. 

As you can see we rather use this instrument program to pass it through to other components. But a realistically seen: It we usually expect everything to be the General Midi instruments.
